File: usedesk_sdk/src/main/java/ru/usedesk/sdk/external/ui/knowledgebase/pages/articlebody/ArticlesBodyFragment.java
package ru.usedesk.sdk.external.ui.knowledgebase.pages.articlebody;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

import java.util.List;

import ru.usedesk.sdk.external.UsedeskKnowledgeBase;
import ru.usedesk.sdk.external.UsedeskSdk;
import ru.usedesk.sdk.external.entity.knowledgebase.ArticleBody;
import ru.usedesk.sdk.external.ui.knowledgebase.common.ViewModelFactory;
import ru.usedesk.sdk.external.ui.knowledgebase.pages.FragmentListView;

public class ArticlesBodyFragment extends FragmentListView<ArticleBody, ArticlesBodyViewModel> {

    public static final String SEARCH_QUERY_KEY = "searchQueryKey";

    private final UsedeskKnowledgeBase usedeskKnowledgeBase;

    public ArticlesBodyFragment() {
        usedeskKnowledgeBase = UsedeskSdk.getUsedeskKnowledgeBase();
    }

    public static ArticlesBodyFragment newInstance(@NonNull String searchQuery) {
        Bundle args = new Bundle();
        args.putString(SEARCH_QUERY_KEY, searchQuery);

        ArticlesBodyFragment articlesBodyFragment = new ArticlesBodyFragment();
        articlesBodyFragment.setArguments(args);
        return articlesBodyFragment;
    }

    @Override
    protected ViewModelFactory<ArticlesBodyViewModel> getViewModelFactory() {
        String searchQuery = getNonNullArguments().getString(SEARCH_QUERY_KEY);

        return new ArticlesBodyViewModel.Factory(usedeskKnowledgeBase, searchQuery);
    }

    @Override
    protected RecyclerView.Adapter getAdapter(List<ArticleBody> list) {
        if (!(getParentFragment() instanceof IOnArticleBodyClickListener)) {
            throw new RuntimeException("Parent fragment must implement " +
                    IOnArticleBodyClickListener.class.getSimpleName());
        }
        return new ArticlesBodyAdapter(list, (IOnArticleBodyClickListener) getParentFragment(),
                usedeskKnowledgeBase.getViewCustomizer());
    }

    public void onSearchQueryUpdate(@NonNull String searchQuery) {
        getViewModel().onSearchQueryUpdate(searchQuery);
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/pages/articlebody/ArticlesBodyFragment.kt
package ru.usedesk.knowledgebase_gui.pages.articlebody

import android.os.Bundle
import android.view.View
import androidx.fragment.app.viewModels
import androidx.lifecycle.LiveData
import androidx.recyclerview.widget.RecyclerView
import ru.usedesk.common_gui.UsedeskBinding
import ru.usedesk.knowledgebase_gui.R
import ru.usedesk.knowledgebase_gui.entity.DataOrMessage
import ru.usedesk.knowledgebase_gui.pages.FragmentListView
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskArticleBodyOld

internal class ArticlesBodyFragment : FragmentListView<UsedeskArticleBodyOld, ArticlesBodyFragment.Binding>(
        R.layout.usedesk_fragment_list,
        R.style.Usedesk_KnowledgeBase
) {

    private val viewModel: ArticlesBodyViewModel by viewModels()

    override fun getAdapter(list: List<UsedeskArticleBodyOld>): RecyclerView.Adapter<*> {
        if (parentFragment !is IOnArticleBodyClickListener) {
            throw RuntimeException("Parent fragment must implement " +
                    IOnArticleBodyClickListener::class.java.simpleName)
        }
        return ArticlesBodyAdapter(list, parentFragment as IOnArticleBodyClickListener)
    }

    override fun init() {
        val searchQuery = argsGetString(SEARCH_QUERY_KEY)
        if (searchQuery != null) {
            viewModel.init(searchQuery)
        }
    }

    override fun createBinding(rootView: View, defaultStyleId: Int) = Binding(rootView, defaultStyleId)

    override fun getLiveData(): LiveData<DataOrMessage<List<UsedeskArticleBodyOld>>> = viewModel.liveData

    fun onSearchQueryUpdate(searchQuery: String) {
        viewModel.onSearchQueryUpdate(searchQuery)
    }

    companion object {
        private const val SEARCH_QUERY_KEY = "searchQueryKey"

        fun newInstance(searchQuery: String): ArticlesBodyFragment {
            return ArticlesBodyFragment().apply {
                arguments = Bundle().apply {
                    putString(SEARCH_QUERY_KEY, searchQuery)
                }
            }
        }
    }

    internal class Binding(rootView: View, defaultStyleId: Int) : UsedeskBinding(rootView, defaultStyleId) {

    }
}